nsot/models.py

```diff
@@ -2161,6 +2167,46 @@ def to_dict(self):
             'resource': resource,
         }
 
+    @property
+    def diff(self):
+        """
+        Return the diff of the JSON representation of the cached copy of a
+        Resource with its current instance
+        """
+        if self.event == 'Create':
+            old = ''
+        else:
+            # Get the Change just ahead of _this_ change because that has the
+            # state of the Resource before this Change occurred.
+            # TODO(nickpegg): Get rid of this if we change the behavior of
+            # Change to store the previous version of the object
+            old_change = Change.objects.filter(
+                change_at__lt=self.change_at,
+                resource_id=self.resource_id,
+                resource_name=self.resource_name
+            ).order_by(
+                '-change_at'
+            )[0]
+            old = json.dumps(old_change._resource, indent=2, sort_keys=True)
+
+        if self.event == 'Delete':
+            current = ''
+        else:
+            resource = globals()[self.resource_name]
+            obj = get_object_or_404(resource, pk=self.resource_id)
```

Comment:
We're mixing view logic here. Model methods shouldn't raise web view errors. We should probably just let the `DoesNotExist` exception bubble up so that it can be handled automatically as a 404 at the API view layer by DRF. Either that or explicitly catch it and raise a `ValidationError`.  We have examples of both in here and I'm conflicted on which to go with.